### PR TITLE
Dev 695 webapp must request past events to microservice

### PR DESCRIPTION
**Pull request by clarkjoao**

### CODE REVIEW

File: src/actions/get-bounty-amount-updated-event.ts [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
 export const author = "clarkjoao";
 
 export default async function action(
   query?: EventsQuery
-): Promise<BountiesProcessedPerNetwork[]> {
-  const bountiesProcessedPerNetwork: BountiesProcessedPerNetwork[] = [];
+): Promise<EventsProcessed> {
+  const eventsProcessed: EventsProcessed = {};
 
   logger.info("retrieving bounty created events");
```
Comment (vhcsilva) [inline src/actions/get-bounty-amount-updated-event.ts:20]:
```suggestion
  logger.info("retrieving bounty updated events");
```

File: src/actions/get-bounty-canceled-event.ts [hunk 2]
```diff
@@ -10,14 +10,14 @@ import logger from "src/utils/logger-handler";
 import { slashSplit } from "src/utils/string";
 
 export const name = "getBountyCanceledEvents";
-export const schedule = "1 * * * * *";
-export const description = "retrieving bounty canceled events";
+export const schedule = "*/30 * * * * *"; // Each 10 minuts
```
Comment (vhcsilva) [inline src/actions/get-bounty-canceled-event.ts:13]: Should this action be executed every 30 seconds or minutes ? I think `*/30 * * * * *` means every 30 seconds

**Review by vhcsilva — COMMENTED** [review]
All right, my only doubt is about the cron time strings, because when using six `*` the first represents the seconds

File: src/actions/get-bounty-canceled-event.ts (hunk 2 above)
Comment (clarkjoao) [reply src/actions/get-bounty-canceled-event.ts]: you right, i will fix it

**Review by vhcsilva — APPROVED** [review]

File: src/actions/get-bounty-canceled-event.ts [hunk 3]
```diff
@@ -10,14 +10,14 @@ import logger from "src/utils/logger-handler";
 import { slashSplit } from "src/utils/string";
 
 export const name = "getBountyCanceledEvents";
-export const schedule = "1 * * * * *";
-export const description = "retrieving bounty canceled events";
+export const schedule = "*/30 * * * *"; // Each 10 minuts
```
Comment (moshmage) [inline src/actions/get-bounty-canceled-event.ts:13]: comment is wrong

File: src/modules/handle-tweet.ts [hunk 4]
```diff
@@ -81,10 +82,35 @@ const events = {
...
+  networkName: string
+) {
+  if (
+    !bounties ||
+    !mainNetworks.includes(networkName.toLocaleLowerCase()) ||
+    !events?.[entity]?.[event]
+  )
+    return;
+
+  const createTweet = Object.values(bounties).map(
```
Comment (moshmage) [inline src/modules/handle-tweet.ts:102]: This must be made _inside_ the `Promise.all` or you wont be taking advantage of the `Promise.all`

```ts
return Promise.all( Object.values( ....
```

Also, _don't_ await inside a promise all -- you want them all to go at the same time, you don't want to await on each loop for the previous to have finished

File: src/modules/handle-tweet.ts [hunk 5]
```diff
@@ -81,10 +82,35 @@ const events = {
...
+    async (item) =>
+      item.bounty &&
```
Comment (moshmage) [inline src/modules/handle-tweet.ts:104]: **PLEASE** use actual if conditions _or_ filter out unwanted results.

File: src/routes/events.routes.ts [hunk 6]
```diff
@@ -42,40 +45,31 @@ const events = {
...
-      );
-      await Promise.all([createTweet]);
-    }
+  if (netoworkName && eventsProcessed[netoworkName]) {
+    await dispatchTweets(
+      eventsProcessed[netoworkName] as BountiesProcessed,
+      entity,
+      event,
+      netoworkName
+    ).catch(console.error);
```
Comment (moshmage) [inline src/routes/events.routes.ts:70]: we need to properly log this, a console.error alone wont suffice

**Review by moshmage — CHANGES_REQUESTED** [review]

File: src/modules/handle-tweet.ts [hunk 7]
```diff
@@ -99,17 +99,18 @@ export async function dispatchTweets(
...
+      .filter((item) => item.bounty)
+      .map((item) =>
+        twitterTweet({
+          entity,
+          event,
+          bountyId: item?.bounty?.issueId as string,
+          networkName,
+        })
+      )
+  ).catch(console.error);
```
Comment (moshmage) [inline src/modules/handle-tweet.ts:113]: this could also be sent to the elastic :)

**Pull request merged**